UltimatePDF_ExternalLogic/BrowserExecution/ODCUltimatePDFExecutionContext.cs

```diff
@@ -24,26 +21,34 @@ internal class UltimatePDFExecutionContext {
             Uri uri, string baseUrl, string locale, string timezone, IEnumerable<CookieParam> cookies,
             ViewPortOptions viewport, PdfOptions options, int timeoutSeconds, Logger logger) {
 
-            logger.Log("Page open...");
-
-            using var pooled = await pool.NewPooledPage(logger);
-            await SetupPage(pooled.Page, uri, viewport, locale, timezone, cookies, timeoutSeconds, logger);
+            logger.Log("Page open... " + uri);
 
-            await pooled.Page.WaitForSelectorAsync(":root:not(.ultimate-pdf-is-not-ready)");
+            Stopwatch sw = new();
+            sw.Start();
 
-            if (!string.IsNullOrEmpty(baseUrl)) {
-                await pooled.Page.EvaluateExpressionAsync("window?.UltimatePDF?.setBaseUrl?.('" + HttpUtility.JavaScriptStringEncode(baseUrl) + "')");
-            }
+            using var pooled = await pool.NewPooledPage(logger);
+            await SetupPage(pooled.Page, uri, viewport, locale, timezone, cookies, 
+                timeoutSeconds, logger, baseUrl);
 
             bool hasLayouts = await pooled.Page.EvaluateExpressionAsync<bool>("window?.UltimatePDF?.hasLayouts?.()");
+            byte[] pdf;
+
             if (hasLayouts) {
-                byte[] pdf = await RenderLayoutPipeline(pooled, logger);
-                return pdf;
+                logger.Log("Using UltimatePDF layout pipeline.");
```

Comment:
is the logger logging details for the shared logs or is this debug stuff that you were using?

Reply:
It's for debugging we don't have a log level implement and the logger only has the .Log